### PR TITLE
resource/aws_ecs_capacity_provider: Add support for instance warmup period

### PR DESCRIPTION
**Pull request by shuheiktgw**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes https://github.com/hashicorp/terraform-provider-aws/issues/16673

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource/aws_ecs_capacity_provider: Add support for instance warmup period
```

Output from acceptance testing:

```
$  make testacc TESTARGS='-run=TestAccAWSEcsCapacityProvider_*'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSEcsCapacityProvider_* -timeout 120m
=== RUN   TestAccAWSEcsCapacityProvider_basic
=== PAUSE TestAccAWSEcsCapacityProvider_basic
=== RUN   TestAccAWSEcsCapacityProvider_disappears
=== PAUSE TestAccAWSEcsCapacityProvider_disappears
=== RUN   TestAccAWSEcsCapacityProvider_ManagedScaling
=== PAUSE TestAccAWSEcsCapacityProvider_ManagedScaling
=== RUN   TestAccAWSEcsCapacityProvider_ManagedScalingPartial
=== PAUSE TestAccAWSEcsCapacityProvider_ManagedScalingPartial
=== RUN   TestAccAWSEcsCapacityProvider_Tags
=== PAUSE TestAccAWSEcsCapacityProvider_Tags
=== CONT  TestAccAWSEcsCapacityProvider_basic
=== CONT  TestAccAWSEcsCapacityProvider_ManagedScalingPartial
=== CONT  TestAccAWSEcsCapacityProvider_ManagedScaling
=== CONT  TestAccAWSEcsCapacityProvider_disappears
=== CONT  TestAccAWSEcsCapacityProvider_Tags
--- PASS: TestAccAWSEcsCapacityProvider_ManagedScalingPartial (66.75s)
--- PASS: TestAccAWSEcsCapacityProvider_ManagedScaling (66.82s)
--- PASS: TestAccAWSEcsCapacityProvider_disappears (68.01s)
--- PASS: TestAccAWSEcsCapacityProvider_basic (85.37s)
--- PASS: TestAccAWSEcsCapacityProvider_Tags (137.94s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	139.787s
```

Thank you for your review! 👍

### CODE REVIEW

**Comment by shuheiktgw** [comment]
This PR conflicts with https://github.com/hashicorp/terraform-provider-aws/pull/16942

File: aws/resource_aws_ecs_capacity_provider.go [hunk 1]
```diff
@@ -63,6 +63,13 @@ func resourceAwsEcsCapacityProvider() *schema.Resource {
 							ForceNew: true,
 							Elem: &schema.Resource{
 								Schema: map[string]*schema.Schema{
+									"instance_warmup_period": {
+										Type:         schema.TypeInt,
+										Optional:     true,
+										Computed:     true,
+										ForceNew:     true,
+										ValidateFunc: validation.IntAtLeast(1),
```
Comment (bflad) [inline aws/resource_aws_ecs_capacity_provider.go:71]: Nit: [API Reference](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_ManagedScaling.html#ECS-Type-ManagedScaling-instanceWarmupPeriod) denotes maximum of 10000

```suggestion
										ValidateFunc: validation.IntBetween(1, 10000),
```

**Review by bflad — COMMENTED** [review]
Looks great, @shuheiktgw, just going to adjust the validation to match API reference. 👍

**Review by bflad — APPROVED** [review]
Thank you, @shuheiktgw 🚀  (Will add changelog entry on merge)

Output from acceptance testing in AWS Commercial:

```
--- PASS: TestAccAWSEcsCapacityProvider_basic (82.59s)
--- PASS: TestAccAWSEcsCapacityProvider_disappears (61.98s)
--- PASS: TestAccAWSEcsCapacityProvider_ManagedScaling (83.84s)
--- PASS: TestAccAWSEcsCapacityProvider_ManagedScalingPartial (60.27s)
--- PASS: TestAccAWSEcsCapacityProvider_Tags (100.69s)
```

Output from acceptance testing in AWS GovCloud (US):

```
--- PASS: TestAccAWSEcsCapacityProvider_basic (64.74s)
--- PASS: TestAccAWSEcsCapacityProvider_disappears (61.46s)
--- PASS: TestAccAWSEcsCapacityProvider_ManagedScaling (56.88s)
--- PASS: TestAccAWSEcsCapacityProvider_ManagedScalingPartial (56.82s)
--- PASS: TestAccAWSEcsCapacityProvider_Tags (126.82s)
```

**Comment by shuheiktgw** [comment]
Thank you for your review! @bflad

**Pull request merged**